source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -981,19 +981,29 @@ OcspStapleAction ServerContextImpl::ocspStapleAction(const CertContext& cert_con
 
 int ServerContextImpl::handleOcspStapling(SSL* ssl, void*) {
   const bool client_ocsp_capable = isClientOcspCapable(ssl);
-  // returns server cert selected for this connection
-  // see https://www.openssl.org/docs/man1.1.1/man3/SSL_set_tlsext_status_ocsp_resp.html
-  // for details
-  auto* selected_cert = SSL_get_certificate(ssl);
-  const auto& cert_context = certificateContext(selected_cert);
-  auto ocsp_staple_action = ocspStapleAction(cert_context, client_ocsp_capable);
+  
+  // Fix OSSM-2208 by looping on all certificates to find at least a good one
+  const CertContext* selected_cert_context = &tls_context_.cert_contexts_[0];
+  auto  ocsp_staple_action = ocspStapleAction(tls_context_.cert_contexts_[0], client_ocsp_capable);
+  for(const auto& cert_context : tls_context_.cert_contexts_) {
+      if(!SSL_select_current_cert(ssl,tls_context_.cert_contexts_[0].cert_chain_.get())) {
```

Comment:
This line is just repeatedly setting the current certificate chain to be the same one from `tls_context_.cert_contexts_[0]` every time, regardless of the value of `cert_context` from the for loop.

I think you want to replace it with: `SSL_select_current_cert(ssl, cert_context.cert_chain_.get())` otherwise you're just selecting the same (first) certificate every time round the loop, regardless of the `cert_context` loop variable.